ee/clickhouse/queries/session_recordings/clickhouse_session_recording_list.py

```diff
@@ -1,119 +1,234 @@
+            filtered_events.timestamp >= session_recordings.start_time
+            AND filtered_events.timestamp <= session_recordings.end_time
+        )
+        {person_id_clause}
+    GROUP BY session_recordings.session_id
+    HAVING 1 = 1
+    {event_filter_aggregate_having_clause}
+    ORDER BY start_time DESC
+    LIMIT %(limit)s OFFSET %(offset)s
     """
 
-    def _get_distinct_id_clause(self) -> Tuple[Dict[str, Any], str]:
-        distinct_id_clause = ""
-        distinct_id_params = {}
-        if self._filter.person_uuid:
-            distinct_id_clause = f"AND distinct_id IN (SELECT distinct_id from person_distinct_id WHERE person_id = %(person_uuid)s AND team_id = %(team_id)s)"
-            distinct_id_params = {"person_uuid": self._filter.person_uuid, "team_id": self._team.pk}
-        return distinct_id_params, distinct_id_clause
-
-    def _get_events_query_with_aggregate_clauses(self) -> EventsQueryWithAggregateClausesSQL:
-        event_filters = format_event_filters(self._filter)
-        events_timestamp_params, events_timestamp_clause = self._get_events_timestamp_clause()
-        event_query = self._event_query.format(
-            events_timestamp_clause=events_timestamp_clause,
-            event_filter_event_select_clause=event_filters.event_select_clause,
-            event_filter_event_where_clause=event_filters.event_where_clause,
-        )
+    def _get_person_query(self,) -> Tuple[str, Dict[str, Any]]:
+        return ClickhousePersonQuery(filter=self._filter, team_id=self._team.pk).get_query()
 
-        params: Dict[str, Any] = {"team_id": self._team.pk, **events_timestamp_params, **event_filters.params}
+    def _has_entity_filters(self):
+        return self._filter.entities and len(self._filter.entities) > 0
 
-        return EventsQueryWithAggregateClausesSQL(
-            event_query, params, event_filters.aggregate_select_clause, event_filters.aggregate_where_clause,
+    def _get_limit(self):
+        return self._filter.limit or self.SESSION_RECORDINGS_DEFAULT_LIMIT
+
+    def _get_person_id_clause(self) -> Tuple[str, Dict[str, Any]]:
+        person_id_clause = ""
+        person_id_params = {}
+        if self._filter.person_uuid:
+            person_id_clause = "AND person.id = %(person_uuid)s"
+            person_id_params = {"person_uuid": self._filter.person_uuid}
+        return person_id_clause, person_id_params
+
+    # We want to select events beyond the range of the recording to handle the case where
+    # a recording spans the time boundaries
+    def _get_events_timestamp_clause(self) -> Tuple[str, Dict[str, Any]]:
+        timestamp_clause = ""
+        timestamp_params = {}
+        if self._filter.date_from:
+            timestamp_clause += "\nAND timestamp >= %(event_start_time)s"
+            timestamp_params["event_start_time"] = self._filter.date_from - timedelta(hours=12)
+        if self._filter.date_to:
+            timestamp_clause += "\nAND timestamp <= %(event_end_time)s"
```

Comment:
Note: We auto-format using sqlparse already so you don't really need to put much effort into prettifying the sql :)